vaadin-date-picker-flow-parent/vaadin-date-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/datepicker/DatePickerLocaleIT.java

```diff
@@ -186,6 +187,54 @@ public void datePickerWithLocale_setInputValue_blur_assertDisplayedValue() {
         assertNoWarnings();
     }
 
+    @Test
+    public void datePickerWithLocale_setInputValue_blur_defaultReferenceDateIsUsed() {
+        LocalDate now = LocalDate.now();
+        int currentYear = now.getYear();
+
+        int testYear1 = (currentYear + 51) % 100;
+        int adjustedYear1 = getAdjustedYear(currentYear, testYear1);
+
+        picker.setInputValue("02/27/" + testYear1);
+        picker.sendKeys(Keys.TAB);
+        Assert.assertEquals("2/27/" + adjustedYear1, picker.getInputValue());
```

Comment:
For parsing tests, using `getInputValue` in the assertion is not a good option, as in case the parsing fails, the input value will just stay the same, but the internal value will not change. It would be better to use `getDate` [as suggested](https://github.com/vaadin/flow-components/pull/3914#discussion_r1019149852).

If some tests are supposed to test the parse -> reformat cycle, then we can do assertions on both.

Reply:
In the newest commit, both assertions are used for the tests.